design/ceph/ceph-orchestrator-osd-add-remove.md

```diff
@@ -0,0 +1,176 @@
+# Adding and removing Ceph osds via Ceph orchestrator interface
+
+## Goals
+
+1. Give Ceph's Rook Orchestrator Plugin the ability to add and remove OSDs to/from a Rook-Ceph
+   cluster running in Kubernetes.
+2. Users of the Ceph Orchestrator CLI and/or Ceph Dashboard should be able to
+   create OSDs on nodes with as much complexity as `ceph-volume` is able to support all possible
+   user needs.
+3. Give as much room as possible for the Ceph Mgr Orchestrator Module and `ceph-volume` to implement
+   OSD management logic so that Rook's code can be simpler and long-term code maintenance on Rook
+   can be reduced.
+
+## Current status
+
+Rook expects to manage storage declaratively. Users can add disks to a Rook cluster by adding them
+to the `CephCluster` resource via the disk name itself, via 2 forms of pattern matching, or via the
+`useAllDisks` option. The Ceph Mgr Orchestrator Module differently expects to add disks to a Ceph
+cluster imperatively (not declaratively) by issuing a `ceph-volume` command with a Ceph
+[Drive Groups](https://github.com/ceph/ceph/blob/master/src/python-common/ceph/deployment/drive_group.py)
+specification. Once the `ceph-volume` command using the Drive Groups spec has been run, Ceph does
+not keep track of the Drive Groups specification used.
```

Comment:
Not sure I understand why the Ceph approach is considered imperative? The DriveGroup is specifically a declarative approach to deploying OSDs. The goal is certainly to have a idempotent call to deploy such a specification (allthough that doesn not work yet)